### PR TITLE
Upgrade pytest version

### LINKED ISSUE #178: Upgrade pytest version
The current version of pytest (6.1.2) results in an error TypeError: required field "lineno" missing from alias, related to [this known issue](https://github.com/pytest-dev/pytest/discussions/9195). This issue was fixed in version >= 6.2.5.

### PR DESCRIPTION
**Pull request by paaige**
Closes #178

### CODE REVIEW

**Review by perlinm — APPROVED** [review]

File: dev-requirements.txt [hunk 1]
```diff
@@ -5,7 +5,7 @@ isort[colors]~=5.10.1
 mypy>=0.961
 nbmake~=1.3.0
 pylint>=2.13.0
-pytest~=6.1.2
+pytest>=6.2.5
 pytest-cov~=2.11.1
 pytest-randomly~=3.10.1
 pytest-socket~=0.4.1
```
Comment (richrines1) [inline dev-requirements.txt:11]: should we change these to `>=` as well?

Comment (paaige) [reply dev-requirements.txt:11]: I guess this only matters if going below these versions break things?

Comment (richrines1) [reply dev-requirements.txt:11]: `~=` should still prevent going below these versions, but does prevent us from using more recent ones (basically `~=2.11.1` is synonymous with `>=2.11.1, <2.12.0`). So we'd be allowing any future update of pytest, but pinning (already out-of date) versions of its plugins

Comment (perlinm) [reply dev-requirements.txt:11]: I vote in favor of `>=` if it doesn't break anything.

**Pull request merged**